pkg/gateway/path/resolver.go

```diff
@@ -15,12 +15,12 @@ const (
 	Separator = "/"
 
 	rePath      = "(?P<path>.*)"
-	reReference = `(?P<ref>\w[-\w]*)`
+	reReference = `(?P<ref>\w[-\w]*[~^0-9]*)`
```

Comment:
Not sure this is a good way to parse here.  TBH I don't understand why we need to enforce morality of references: why not just take *everything* up to the next slash and try to use that?  IIRC this was originally due to using the parsed branch name in a DB lookup, with the (misguided but common) fear that maybe Bobbie Tables can overcome DB parametrized queries and perform SQL injection.  I say: just allow all characters here, and fail when you revparse them!